Refl Preview: Replace Image Info On RegionSelector with Statusbar

cailafinn opened the pull request:
**Description of work.**
Removed the ImageInfoWidget from the regionselector and allowed it to be passed in instead. This means that we can pass in a much more minimal version that sits at the bottom of the tab. Future work will hopefully allow this to be used for all three of the widgets on the Preview Tab. 

**To test:**

1. Open the ISIS Reflectometry interface
2. Go to the Preview tab
3. Enter run `INTER45455_inst` and click Load (This data is available on this Epic and must be in your user search path: #31069)
4. Check that the region selector doesn't have the table or dimension selector on it any more (open the sliceviewer for reference)
5. Check that the statusbar at the bottom of the tab is being updated with the X and Y values from the regionselector. 

Fixes #34671 


<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

Commit:
Remove Qt types from interface

This will let us be a bit more fast and loose with what counts as an
ImageInfoWidget. We want to be able to pass in a non-table type so that
we can have a statusbar instead.

RE #34671

Commit:
Insert the InfoWidget as a parameter

RE #34671

Commit:
Create mini imageinfowidget

This is just a QLabel for now. We can shrink down what it displays
later.

RE #34671

Commit:
Allow injection of imageinfowidget into regionselector

RE #34671

Commit:
Add option to pass miniinfo through to python

Adds a wrap helper function to our sip import that allows the wrapping
of a C++ object into a python object, provided that the object is
represented in the _common.sip file.

RE #34671

Commit:
Add widget to GUI

RE #34671

Commit:
Simplify mini info widget to only care about X and Y

RE #34671

Commit:
Also hide dimentions changer when in Refl GUI

RE #34671

Commit:
Fix warning about unused param

RE #34671

Commit:
Fix test build failures

RE #34671

Commit:
Fix docstring's missing parameter.

RE #34671

Commit:
Fix test failing when not finding sip link

The tests don't seem to be able to find the type of the
ImageInfoWidgetMini. So add a check to make sure we don't segfault at
least.

REEEEEE #34671

gemmaguest commented on qt/scientific_interfaces/ISISReflectometry/GUI/Preview/QtPreviewView.cpp:
```diff
@@ -21,16 +21,19 @@
 namespace MantidQt::CustomInterfaces::ISISReflectometry {
-QtPreviewView::QtPreviewView(QWidget *parent) : QWidget(parent) {
+QtPreviewView::QtPreviewView(QWidget *parent)
+    : QWidget(parent), m_imageInfo(new MantidQt::MantidWidgets::ImageInfoWidgetMini(this)) {
```
Just double checking that this is correct to be creating a raw pointer with `new`? I think so because as it is a QWidget then Qt will destroy it when the parent is destroyed - is that your understanding too?

ConorMFinn replied:
I think so, yes. That's the way we've handled adding QWidgets in other parts of the GUI, so I've just followed that pattern

gemmaguest committed:
Add new field to statusbar and clear if no values

- Adds Signal to the info displayed by the mini info widget
- Clear the widget if values are empty (assume if the first value is
  empty, they all are).

Re #34671

gemmaguest committed:
Move interface header out of moc files now it's no longer a QWidget

This was causing a compiler warning "no relevant classes found"

Re #34671

gemmaguest reviewed (dismissed):
Code changes look good and this works really well - the new statusbar looks a lot neater in this  GUI than the original image info widget :+1:. There is a little more logic in the view than I'd like but this follows the same design as the original ImageInfoWidget so I think is the best solution for now for consistency. 

![image](https://user-images.githubusercontent.com/12895056/209148159-e2cd9acf-9903-441e-a3d5-80e575cd0e1a.png)

Since @ConorMFinn is away I added a couple of minor fixes:
- Include "Signal" in the values displayed.
- Clear values when not hovering over the plot (which was displaying "-" previously).
- Move the header file for the interface out of the MOC files section, now it is no longer a QWidget.

I also rebased onto main.

gemmaguest committed:
Add virtual destructor

Re #34671

gemmaguest reviewed (approved):
The latest change just added a virtual destructor to fix a Jenkins complaint. This all looks good to me now.

martyngigg commented:
> Code changes look good and this works really well - the new statusbar looks a lot neater in this GUI than the original image info widget 👍. There is a little more logic in the view than I'd like but this follows the same design as the original ImageInfoWidget so I think is the best solution for now for consistency.
> 
> ![image](https://user-images.githubusercontent.com/12895056/209148159-e2cd9acf-9903-441e-a3d5-80e575cd0e1a.png)
> 
> Since @ConorMFinn is away I added a couple of minor fixes:
> 
>     * Include "Signal" in the values displayed.
> 
>     * Clear values when not hovering over the plot (which was displaying "-" previously).
> 
>     * Move the header file for the interface out of the MOC files section, now it is no longer a QWidget.
> 
> 
> I also rebased onto main.

I do like how clean the information is presented here. Perhaps the sliceviewer would look better with it presented like this 🤔
